### PR TITLE
Add command for redacting backups

### PR DESCRIPTION
**Pull request by josephschorr**
Helpful for providing community support without leaking the names, comments and/or object IDs in the permission system

Related: #194

### CODE REVIEW

**Review by vroldanbet — COMMENTED** [review]
LGTM, just a comment

Can we 🙏🏻  merge https://github.com/authzed/zed/pull/316 first? It's had its fair share of rebasing already

File: pkg/backupformat/redaction.go [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+	Definitions map[string]string
+
+	// Caveats is the map of original caveat names to their redacted names.
+	Caveats map[string]string
+
+	// Relations is the map of original relation names to their redacted names.
+	Relations map[string]string
+
+	// ObjectIDs is the map of original object IDs to their redacted names.
+	ObjectIDs map[string]string
```
[the remark this answers is not in the document]
Comment (josephschorr) [reply pkg/backupformat/redaction.go:40]: My concern with using a hash is the risk of conflicts

File: pkg/backupformat/redaction.go [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+		}
+
+		for _, caveatDef := range compiled.CaveatDefinitions {
+			if opts.RedactDefinitions {
+				redactionMap.Caveats[caveatDef.Name] = "cav" + strconv.Itoa(redactionCount)
+				redactionCount++
+				caveatDef.Name = redactionMap.Caveats[caveatDef.Name]
+			}
+
+			// TODO: Redact caveat parameters.
```
[the remark this answers is not in the document]
Comment (josephschorr) [reply pkg/backupformat/redaction.go:169]: Gotcha

**Comment by josephschorr** [comment]
Rebased

**Review by vroldanbet — APPROVED** [review]

**Pull request merged**